### PR TITLE
[WIP] Finetuning MVP script

### PR DESCRIPTION
**Pull request by nancyhung**
This change creates a more generic data conversion script to convert user-input data into MDS format.

We plan to productionize this script into an mcli command and a separate mcli sdk endpoint that users can easily call to start a finetuning workflow.

### CODE REVIEW

File: examples/end-to-end-examples/finetuning/finetune_small_dataset.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    #     cluster_name=CLUSTER_NAME,
+    #     checkpoint_path=f'{SAVE_FOLDER}/latest-rank0.pt.symlink',
+    #     hugging_face_path=HUGGING_FACE_PATH,
+    #     gpu_num=8,
+    #     gpu_type=CLUSTER_TO_GPU_MAP[CLUSTER_NAME],
+    # )
+
+    # check_run_status_and_raise_if_error(conversion_run)
+
+    # print(f'Checkpoint converted to Hugging Face format and saved to {HUGGING_FACE_PATH}')
```
Comment (nancyhung) [inline examples/end-to-end-examples/finetuning/finetune_small_dataset.py:247]: fyi i still need to clean this up

File: examples/end-to-end-examples/finetuning/finetune_small_dataset.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        output_folder='oci://nancy-test',
+        train_index_path='oci://nancy-test/10k_train.jsonl',
+        eval_index_path='oci://nancy-test/10k_validate.jsonl',
+        concat_tokens=0,
+        eos_text='</s>',
+        bos_text='<s>',
+        no_wrap=True,
+        max_workers=1,
+        compression='gzip'
+    )
```
Comment (nancyhung) [inline examples/end-to-end-examples/finetuning/finetune_small_dataset.py:214]: have not tested this part yet -- the train_index_path/eval_index_path are incorrect right now

File: examples/end-to-end-examples/finetuning/finetune_small_dataset.py [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+HUGGING_FACE_PATH: str = 'oci://nancy-test/checkpoints/hf-path-08-07-23::04:06:08'
+INFERENCE_CLUSTER_NAME = 'r7z14'
+NUM_REPLICAS: int = 1
+
+# TODO: remove, this is for testing
+run_name = f'finetune-{MODEL_NAME.value}'
+CLUSTER_TO_GPU_MAP = {
+    'r1z1': 'a100_80gb',
+    'r7z2': 'a100_40gb',
+}
```
Comment (nancyhung) [inline examples/end-to-end-examples/finetuning/finetune_small_dataset.py:33]: will remove this though maybe useful for customers to fill out?

File: examples/end-to-end-examples/finetuning/utils_convert_to_mds.py [hunk 4]
```diff
@@ -0,0 +1,269 @@
+# Copyright 2022 MosaicML Examples authors
```
Comment (dakinggg) [inline examples/end-to-end-examples/finetuning/utils_convert_to_mds.py:1]: I think this script actually belongs in foundry with the other data prep scripts: https://github.com/mosaicml/llm-foundry/tree/main/scripts/data_prep

**Comment by nancyhung** [comment]
Closing because we want to move this to foundry.

**Pull request closed without merging**